recipe-client-addon/lib/ClientEnvironment.jsm

```diff
@@ -203,6 +203,10 @@ this.ClientEnvironment = {
       return Utils.keyBy(addons, "id");
     });
 
+    XPCOMUtils.defineLazyGetter(environment, "isFirstRun", () => {
```

Comment:
Are you deliberately not using `XPCOMUtils.defineLazyPreferenceGetter()` to avoid the observer, or...?